### PR TITLE
Add unit tests for settings

### PR DESCRIPTION
**Pull request by Neverlord**
I noticed that the `settings` header had no unit tests. This test suite covers all functions defined in `settings.hpp`.

### CODE REVIEW

File: libcaf_core/test/settings.cpp [hunk 1]
```diff
@@ -0,0 +1,162 @@
...
+  CAF_CHECK(x.contains("logger"));
+  CAF_CHECK(x.contains("one"));
+  CAF_CHECK_EQUAL(unpack(x, "foo"), "bar"_s);
+  CAF_CHECK_EQUAL(unpack(x, "logger", "console"), atom("none"));
+  CAF_CHECK_EQUAL(unpack(x, "one", "two", "three"), "four"_s);
+  put(x, "logger.console", atom("trace"));
+  CAF_CHECK_EQUAL(unpack(x, "logger", "console"), atom("trace"));
+}
+
+CAF_TEST(put missing) {
```
Comment (jakobod) [inline libcaf_core/test/settings.cpp]: You are not checking that an already contained value isn't added again! Maybe try to add some contained value with a different key and check that the size stays the same and the old key + value is still contained afterwards.

Comment (Neverlord) [reply libcaf_core/test/settings.cpp]: The last `put_missing` checks that, does it not?

```c++
  put_missing(x, "logger.console", atom("trace"));
  CAF_CHECK_EQUAL(unpack(x, "logger", "console"), atom("none"));
```

Comment (jakobod) [reply libcaf_core/test/settings.cpp]: True. I misread `atom("trace")` for `atom("none")`.
I'm still confused by the description of the function regarding this test:
```
/// Converts `value` to a `config_value` and assigns it to `key` if `value` is
/// currently missing in `xs`.
```
Shouldn't you try to add the same value with a different key? The description doesn't say anything about handling same keys.

Comment (Neverlord) [reply libcaf_core/test/settings.cpp]: Documentation fail. Will fix.

**Pull request merged**